lib/public_key/src/pubkey_cert.erl

```diff
@@ -1805,41 +1805,48 @@ compatible_ext_key_usage(undefined, _, endentity) -> %% keyusage (first arg )is
     true;
 compatible_ext_key_usage(_, undefined, _) ->
     true;
-compatible_ext_key_usage(#'Extension'{extnValue = KeyUse}, #'Extension'{extnValue = Purposes}, _) ->
+compatible_ext_key_usage(#'Extension'{extnValue = KeyUse}, #'Extension'{extnValue = Purposes}, Type) ->
```

Comment:
maybe improve the readability by having variable names in clause head more related to extension names.

- Purposes -> ExtKeyUse

(similarly to other places in module)

Reply:
I think it helps with having a good help function name that makes sense so  I added an assertion of the extesion-ids in the function clause  and made KeyUse -> KeyUses. I think it can help.